### ISSUE TITLE
Mac OS X packages do not include source files

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#2728]
This would be very useful, though, so that one can get proper backtraces in a debugger, and see the condition that fails and the variables that led to it.

@luca-heltai -- opinions?

**Comment by davydden** [comment]
@bangerth AFAIK the latest macOS build uses Homebrew underneath, which in turn supports only temporary folders during compilation that are deleted after successful build. Same applies to Spack, where you could keep the build folder but it would still be in some temporary location like `/private/tmp/var/xyz/5se/`, determined by OS.

**Comment by bangerth** [comment]
Oh, I see. That's a pity :-( I think the temporary location would not be a problem (who cares where the file is, as long as the debugger can find it), though that may not be pretty.

Any other ideas?

**Comment by davydden** [comment]
temporary location would not be a problem if you compile yourself (as what i do with Spack), but it might be a problem for packaging dealii suite. Either way, i am sure @luca-heltai will have more details.

**Comment by Rombur** [comment]
Actually temporary location is a "problem" with Spack. For me, the temporary location is /tmp which is deleted after a reboot. It's not really a problem for me since I compile deal myself though.

**Comment by davydden** [comment]
right, it is in `/tmp` and therefore will stay alive only untill reboot.

**Comment by luca-heltai** [comment]
Actually, the .dmg is constructed by using brew for all packages *except* deal.II. Adding source files should not be a problem. Do you want the full git repository, or just the latest source files?

**Comment by davydden** [comment]
@Rombur a bit off-topic, but one can actually keep sources in spack outside of temp folder https://github.com/LLNL/spack/issues/1137#issuecomment-229290024

**Comment by Rombur** [comment]
@davydden thanks

**Comment by luca-heltai** [comment]
I'm upgrading all packages with brew, and I'll upload a new version soon, that includes source files. The latest one that I uploaded here

https://github.com/luca-heltai/dealii/releases/tag/v8.5.pre.1

seems to have some issues on my pc, and I'm trying to figure out what the problem may be...

**Comment by bangerth** [comment]
@luca-heltai -- I don't think it's necessary to have the full repository. That's also a lot of data. I just want the source files so you can backtrace into the code in a debugger.

**Comment by luca-heltai** [comment]
The problem on the above image was the same as in issue #2739

**Comment by luzpaz** [comment]
Is this still relevant ?

Asking because I'm wondering if there is an ability to create a homebrew recipe for dealii ? (https://github.com/dealii/dealii/wiki/MacOSX)
